demo/admin/src/products/categories/ProductsSelectGrid.tsx

```diff
@@ -0,0 +1,272 @@
+            renderCell: (params) => {
+                return (
+                    <CrudVisibility
+                        visibility={params.row.status == "Published"}
+                        onUpdateVisibility={async (status) => {
+                            await client.mutate<GQLUpdateProductStatusMutation, GQLUpdateProductStatusMutationVariables>({
+                                mutation: updateProductStatusMutation,
+                                variables: { id: params.row.id, status: status ? "Published" : "Unpublished" },
+                                optimisticResponse: {
+                                    __typename: "Mutation",
+                                    updateProduct: { __typename: "Product", id: params.row.id, status: status ? "Published" : "Unpublished" },
+                                },
+                            });
+                        }}
+                    />
+                );
+            },
+        },
+    ];
+
+    const { filter: gqlFilter, search: gqlSearch } = muiGridFilterToGql(columns, dataGridProps.filterModel);
+
+    const { data, loading, error } = useQuery<GQLProductsListQuery, GQLProductsListQueryVariables>(productsQuery, {
+        variables: {
+            filter: gqlFilter,
+            search: gqlSearch,
+            offset: dataGridProps.page * dataGridProps.pageSize,
+            limit: dataGridProps.pageSize,
+            sort: muiGridSortToGql(sortModel, dataGridProps.apiRef),
+        },
+    });
+    const rows = data?.products.nodes ?? [];
+    const rowCount = useBufferedRowCount(data?.products.totalCount);
+
+    return (
+        <DataGridPro
+            {...dataGridProps}
+            disableSelectionOnClick
+            rows={rows}
+            rowCount={rowCount}
+            columns={columns}
+            loading={loading}
+            error={error}
+            components={{
+                Toolbar: ProductsGridToolbar,
+            }}
+            checkboxSelection
+            keepNonExistentRowsSelected
+            selectionModel={input.value ? input.value : []}
+            onSelectionModelChange={(newSelectionModel) => {
+                input.onChange(newSelectionModel);
+            }}
+        />
+    );
+}
+
+const productsFragment = gql`
+    fragment ProductsListManual on Product {
+        id
+        slug
+        title
+        description
+        price
+        type
+        additionalTypes
+        inStock
+        image
+        status
+        category {
+            id
+            title
+        }
+        tags {
+            id
+            title
+        }
+        colors {
+            name
+            hexCode
+        }
+        variants {
+            id
+        }
+        articleNumbers
+        discounts {
+            quantity
+            price
+        }
+    }
+`;
+
+const productsQuery = gql`
+    query ProductsList($offset: Int!, $limit: Int!, $sort: [ProductSort!], $filter: ProductFilter, $search: String) {
```

Comment:
GraphQL operation names need to be unique across the project, and this names are already used in the generated products grid. This can lead to problems with caching and/or generation once the queries differ (e.g., you remove a field here). Please rename the operations in this file.

Reply:
> This can lead to problems with caching

I don't think the query name is used in apollo cache. After all it's a normalisation across all queries, with cache-key type+id

> generation once the queries differ

since we switched to near-operations-file in gql-codegen those queries can have the same name if they are used in different files. The types are generated per file, so there are no conflicts in naming.

----

The only reason for unique query names might be for profiling or debugging on the server side (so you get a hint where the query comes from)

(NOTE: fragments have to be unique, I don't know why though)